meilisearch/src/routes/indexes/documents.rs

```diff
@@ -227,26 +231,51 @@ async fn document_addition(
 
     let (uuid, mut update_file) = index_scheduler.create_update_file()?;
 
-    // TODO: this can be slow, maybe we should spawn a thread? But the payload isn't Send+Sync :weary:
-    // push the entire stream into a `Vec`.
-    // If someone sends us a never ending stream we're going to block the thread.
-    // TODO: Maybe we should write it to a file to reduce the RAM consumption
-    // and then reread it to convert it to obkv?
-    let mut buffer = Vec::new();
+    let temp_file = match tempfile() {
+        Ok(temp_file) => temp_file,
+        Err(e) => {
+            return Err(MeilisearchHttpError::Payload(ReceivePayloadErr(Box::new(e))));
+        }
+    };
+
+    let mut buffer = File::from_std(temp_file);
```

Comment:
> Oops, also second big comment, for the async part we should definitely use an AsyncBufWriter or we're going to have terrible performances: https://docs.rs/tokio/latest/tokio/io/struct.BufWriter.html !

I think we lost this change? Or I forgot about it and approved the last time :grimacing: 

```suggestion
    let async_file = File::from_std(temp_file);
    let mut buffer = BufWriter::new(async_file);
```

Reply:
Already processed